sources/exec_text_tree.c

```diff
@@ -43,8 +43,8 @@ static int	exec_rec(t_param *param, t_treenode *node)
 		node = node->sibling_next->sibling_next;
 	while (!res && node && !is(node->content, "||") && node->sibling_next)
 		node = node->sibling_next->sibling_next;
-	if (!node || !node->sibling_next)
-		return (res);
+	if (param->exiting || !node || !node->sibling_next)
+		return (ft_if_i(param->exiting, 0, res));
```

Comment:
is param-exiting for signaling about command exit? is it temporal solution and then the program will exit from our simulated shell?

Reply:
This is a strategic solution, this is an indicator that the whole program is starting to shut down and nothing new should be initiated. Probably could have implemented this with a special errno, but this looks sufficient